enhancements/machine-config/on-cluster-builds.md

```diff
@@ -0,0 +1,1009 @@
+pools will need to be rebuilt. Where that might be a little tricky is on
+single-node OpenShift (SNO) deployments. We could detect that and
+constrain it only to a single build.
+
+The `customContent` struct could also be injected into the InstallConfig
+schema on a per-pool basis as well as on a global basis to enable Day 1
+configs with custom OS content. However, the struct would have to be
+extended slightly to permit the actual pull secrets to be injected as
+well as to provide some control over the name of the secrets. These
+secrets would have to live within the MCO namespace.
+
+Additionally, the following annotations will be exposed to the cluster
+admin to configure various behaviors of the build and image rollout
+process on a per-pool basis:
+
+| **Annotation Name:**                             | **Description:**                                                                                                                                           |
+| ------------------------------------------------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `machineconfiguration.openshift.io/rebuildImage` | When present, this will instruct the BuildController to clear any failed image builds (and any supporting transitory objects created) and retry the build. |
+| `machineconfiguration.openshift.io/pauseRollout` | When set, the build and push process will occur as usual. However, the NodeController will not roll out the new image until this annotation is cleared.    |
+
+#### ConfigMaps (corev1.ConfigMap)
+
+Absent a more complete specification for bootc’s ConfigMap support,
+let’s assume for the sake of this enhancement that bootc will respect
+the annotations identified here for the ConfigMaps it consumes:
+
+| **Annotation Name:**                                  | **Description:**                                                                                                                                                                                                                                                                                              |
+| ----------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `bootc.coreos.io/configMap`                           | Identifies this ConfigMap as being one that bootc can understand. This could enable cluster admins to create ConfigMaps in the MCO namespace in the future to be directly consumed by bootc. Alternatively, the MCO could detect the presence of the annotations below this.                                  |
```

Comment:
This section is just me forming an opinion around what annotations bootc could recognize / understand from a given ConfigMap. It would be handy if bootc ends up supporting a more advanced CRD for this purpose. Ultimately, this would be in the bootc teams wheelhouse.

Here's kind of a hypothetical question: In addition to ConfigMaps, what if bootc could also understand MachineConfigs / Ignition? The MCO already knows how to combine multiple MachineConfigs into a singular rendered MachineConfig.